raml-parser-2/src/main/java/org/raml/v2/internal/impl/commons/model/type/TypeDeclaration.java

```diff
@@ -90,6 +89,42 @@ public String type()
         return getTypeExpression(getTypeNode());
     }
 
+    public List<TypeDeclaration> parentTypes() {
+
+        List<TypeDeclaration> nodes =  new ArrayList<>();
+        Set<String> parentNames = new HashSet<>();
+        Node parents =  resolvedType.getTypeDeclarationNode().get("type");
+        if ( parents == null) {
+
+           return nodes;
+        }
+
+        List<Node> parentTypeNodes = parents.getChildren();
+        for (Node parentTypeNode : parentTypeNodes) {
+            NamedTypeExpressionNode parent = (NamedTypeExpressionNode) parentTypeNode;
+            parentNames.add(parent.getValue());
+        }
+
+        Node nn = node.getRootNode().get("types");
+        if ( nn == null ) {
+            return nodes;
+        }
+
+        TypeDeclarationModelFactory factory = new TypeDeclarationModelFactory();
```

Comment:
I was more thinking in something like this. This is a pseudo code but I think it shows the way
```
    public List<TypeDeclaration<?>> superTypes() {
        List<TypeDeclaration<?>> result = new ArrayList<>();
        Node typeNode = getTypeNode();
        if (typeNode instanceof ArrayNode) {
            List<Node> children = typeNode.getChildren();

            for (Node child : children) {
                result.add(toTypeDeclaration(child));
            }
        } else {
            result.add(toTypeDeclaration(typeNode));
        }
        return result;
    }

    protected  TypeDeclaration<?> toTypeDeclaration(Node typeNode){
        return new TypeDeclarationModelFactory().create(typeNode);
    }
```